Drivers/STM32H7xx_HAL_Driver/Inc/stm32h7xx_hal_gpio.h

```diff
@@ -286,7 +282,25 @@ void HAL_GPIO_EXTI_Callback(uint16_t GPIO_Pin);
 /** @defgroup GPIO_Private_Constants GPIO Private Constants
   * @{
   */
-
+#define GPIO_MODE_Pos                           0u
+#define GPIO_MODE                               (0x3uL << GPIO_MODE_Pos)
+#define MODE_INPUT                              (0x0uL << GPIO_MODE_Pos)
+#define MODE_OUTPUT                             (0x1uL << GPIO_MODE_Pos)
+#define MODE_AF                                 (0x2uL << GPIO_MODE_Pos)
+#define MODE_ANALOG                             (0x3uL << GPIO_MODE_Pos)
+#define OUTPUT_TYPE_Pos                         4u
+#define OUTPUT_TYPE                             (0x1uL << OUTPUT_TYPE_Pos)
+#define OUTPUT_PP                               (0x0uL << OUTPUT_TYPE_Pos)
+#define OUTPUT_OD                               (0x1uL << OUTPUT_TYPE_Pos)
```

Comment:
This is the location of the offending macro that causes name-collision

Reply:
Since macros are basically just global find-and-replace text replacement, the preprocessor is inserting it into the enum regardless of the code (and scoping) around it :/

Reply:
Ah, yup! You're totally right. I forgot that macros just do a brute-force text replace everywhere. This is totally fine.
Thanks for the added detail.